### PR TITLE
Support STS credentials for Namespace store

### PR DESCRIPTION
**Pull request by liranmauda**
### Explain the changes
1. Adding support for Namespace STS
2. Adding CLI for creating NS STS
3. Changing the validation to allow STS on NS

### CODE REVIEW

**Comment by liranmauda** [comment]
This PR is still in draft as it was not tested.
@ezio-auditore can you also review?

File: pkg/backingstore/reconciler.go [hunk 1]
```diff
@@ -257,7 +257,7 @@ func (r *Reconciler) ReconcilePhases() error {
 
 // LoadBackingStoreSecret loads the secret to the reconciler struct
 func (r *Reconciler) LoadBackingStoreSecret() error {
-	if !util.IsSTSClusterBS(r.BackingStore) {
+	if !util.IsSTSClusterBS(r.BackingStore) {  //LMLM TODO: why is this negative condition, why not return nill if it is true?
```
Comment (ezio-auditore) [inline pkg/backingstore/reconciler.go:260]: If it is not an sts cluster and the secret is also not found it should return an error, And if it is an STS cluster it should not check for the secret.

**Review by ezio-auditore — COMMENTED** [review]
Besides some minor comments, LGTM

File: pkg/backingstore/reconciler.go (hunk 1 above)
Comment (liranmauda) [reply pkg/backingstore/reconciler.go:260]: The thing with negative conditions is that it is harder for humans to understand... 

The translation to a positive statement is in the second half of your sentence: 
> if it is an STS cluster it should not check for the secret.
Which can in turn be written like:
```
	if util.IsSTSClusterBS(r.BackingStore) {
		return nil
	}
```
@ezio-auditore  WDYT?

Comment (ezio-auditore) [reply pkg/backingstore/reconciler.go:260]: ok, makes sense.

**Comment by dannyzaken** [comment]
pending on core changes and @liranmauda is trying to set up an env

**Comment by liranmauda** [comment]
> pending on core changes and @liranmauda is trying to set up an env

https://github.com/noobaa/noobaa-core/pull/7074
@dannyzaken @romayalon Please review.

**Review by jackyalbo — APPROVED** [review]

**Pull request merged**